src/componants/Articles.jsx

```diff
@@ -1,17 +1,26 @@
-import { useEffect, useState } from "react";
+import { useContext, useEffect, useState } from "react";
 import { getArticleById, getArticles, getRandomArticleId } from "../../api";
 
 import {ArticleCard, RandomArticleCard}from "./ArticleCard";
+import { useParams } from "react-router-dom";
+import { useCol } from "react-bootstrap/esm/Col";
+import categoryContext from "../contexts/categoryContexts";
 
 function Articles() {
 	const [articles, setArticles] = useState([]);
 	const [randomArticle, setRandomArticle] = useState({});
+  const {topic} = useParams()
+  const [loaded, setLoaded] = useState(false)
+  const {setCategory} = useContext(categoryContext)
```

Comment:
You could do this without having it context. Do you need the category state to be available everywhere in your app?